shell/common/shell.cc

```diff
@@ -1068,11 +1068,16 @@ void Shell::OnPlatformViewSetNextFrameCallback(const fml::closure& closure) {
 }
 
 // |Animator::Delegate|
-void Shell::OnAnimatorBeginFrame(fml::TimePoint frame_target_time,
+void Shell::OnAnimatorBeginFrame(fml::TimePoint vsync_start_time,
+                                 fml::TimePoint frame_target_time,
                                  uint64_t frame_number) {
   FML_DCHECK(is_setup_);
   FML_DCHECK(task_runners_.GetUITaskRunner()->RunsTasksOnCurrentThread());
 
+  display_manager_->UpdateRefreshRateIfNecessary(
+      flutter::DisplayUpdateType::kUpdateRefreshRate, vsync_start_time,
+      frame_target_time);
```

Comment:
What's the cost of this on iOS? We know for sure that on some Android platforms it's pretty expensive (e.g. 2-5ms) and this is during frameworkload.

Can we instead listen to refresh rate updates from the OS somehow?

Reply:
Could you elaborate how this is expensive on Android. This method is essentially a no-op if framerate doesn't change. See:  https://github.com/flutter/engine/pull/30054/files#diff-c9141b5122b7cb9dcc29a05b1660dbb92c1c6ea8e0df831c96b42708fa1e20b5R66
(I should probably need to update the code to return early so it is clearer)